tests/test/sgx/test_sgx_module.cpp

```diff
@@ -0,0 +1,146 @@
+#include <catch2/catch.hpp>
+
+#include <faabric/scheduler/Scheduler.h>
+#include <sgx/SGXWAMRWasmModule.h>
+
+namespace tests {
+class SgxModuleTestFixture
+{
+  public:
+    SgxModuleTestFixture()
+    {
+        msg = faabric::util::messageFactory("demo", "hello");
+        msg.set_issgx(true);
+        otherMsg = faabric::util::messageFactory("demo", "echo");
+        otherMsg.set_issgx(true);
+    }
+
+    ~SgxModuleTestFixture()
+    {
+        module.reset(msg, "");
+        otherModule.reset(otherMsg, "");
+    }
+
+  protected:
+    wasm::SGXWAMRWasmModule module;
+
+    wasm::SGXWAMRWasmModule otherModule;
+
+    faabric::Message msg;
+
+    faabric::Message otherMsg;
+};
+
+TEST_CASE_METHOD(SgxModuleTestFixture,
+                 "Test binding and unbinding to function",
+                 "[sgx]")
+{
+    // We can unbind when nothing is bound
+    REQUIRE_NOTHROW(module.reset(msg, ""));
+
+    // Try binding to a function
+    REQUIRE_NOTHROW(module.doBindToFunction(msg, false));
+
+    // Then unbind
+    REQUIRE_NOTHROW(module.reset(msg, ""));
```

Comment:
Same point re. all of these `REQUIRE_NOTHROWS`, it's implicit that a test will fail if there's an exception. You can add a comment to say "This test is just checking that no exception is thrown".